modules/jolt_physics/objects/jolt_shaped_object_3d.cpp

```diff
@@ -167,7 +167,14 @@ void JoltShapedObject3D::_update_shape() {
 		return;
 	}
 
-	space->get_body_iface().SetShape(jolt_id, jolt_shape, false, JPH::EActivation::DontActivate);
+	JPH::BodyInterface &iface = space->get_body_iface();
+
+	if (uses_shape_materials()) {
+		// This body/area uses per-shape physics materials, manifold reduction is not safe (can merge shapes with different materials).
+		iface.SetUseManifoldReduction(jolt_id, false);
```

Comment:
You may want to consider turning manifold reduction ON again if the shape no longer uses shape materials (and if `!reports_contacts()`). This saves some CPU cycles during collision detection.